enterprise/frontend/src/metabase-enterprise/content_translation/utils.ts

```diff
@@ -133,9 +134,34 @@ export const translateFieldValuesInSeries = (
     }
     const untranslatedRows = singleSeries.data.rows.concat();
 
-    const translatedRows = singleSeries.data.rows.map((row) =>
-      row.map((value) => tc(value)),
-    );
+    let translatedRows: RowValue[][];
+
+    if (singleSeries.card?.display === "pie") {
+      const pieRows = singleSeries.card.visualization_settings?.["pie.rows"];
+      const keyToNameMap = pieRows
+        ? pieRows.reduce(
+            (acc, row) => {
+              acc[row.key] = row.name;
+              return acc;
+            },
+            {} as Record<string, string>,
+          )
+        : {};
+
+      translatedRows = singleSeries.data.rows.map((row) =>
+        row.map((value) => {
+          if (typeof value === "string" && keyToNameMap[value] !== undefined) {
+            return tc(keyToNameMap[value]);
```

Comment:
Would the row value be a translation key in this case? I think a little comment would be nice, since I didn't fully get why this is the case.